### DISCUSSION TITLE
Codespaces custom subdomain support?

### DISCUSSION THREAD

**Discussion by davidronk** [community/community#8665] [Codespaces]
Are there plans to add custom subdomain support to codespaces?
Or is there a way to do it now?

Our app currently determines the tenant via the subdomain: `tenant.app.com` so it would be great if we could define (or allow *) subdomains for the public forwarded ports that codespaces exposes.  
So in addition to: `<user>-<org>-<repo>-<code>-<port>.githubpreview.dev`
we could also have: `<tenant>.<user>-<org>-<repo>-<code>-<port>.githubpreview.dev`

**Comment by asciimike** [comment]
I don't believe there is a way to handle this at present, and I don't think we could easily handle it (I assume signing certs for `*.<user>-<org>-<repo>-<code>-<port>.githubpreview.dev` is a pain, though maybe https://letsencrypt.org/docs/faq/#does-let-s-encrypt-issue-wildcard-certificates solves that?). 

I will say we have considered custom domain support (so you could run `<user>-<org>-<repo>-<code>-<port>.company.tld`, though this isn't quite what you're looking for, and is potentially an easier problem to solve on our side).

**Reply by billiegoose** [reply]
Just want to chime in we at [Stoplight.io](https://github.com/stoplightio/) were also testing out CodeSpaces when they first came out, and subdomains proved our stumbling block. The first thing a new user does is create a "workspace" with our product, like `your-workspace.stoplight-local.com`. I'm tempted to re-engineer all our routing to support an alternate code path that doesn't relyl on subdomains _just_ so we can use CodeSpaces, but it would be wonderful to support the tenant pattern so well-described by davidronk above out-of-the-box.

**Reply by Chuxel** [reply]
If you are using VS Code locally, you can forward the port to localhost and then update your hosts file to simulate this. If you've got your own SSL certificate, you can leave the port type as blank (or HTTP) and that cert will come across.  You can also disable this setting in your VS Code settings if you prefer.

![image](https://user-images.githubusercontent.com/312252/156397526-0053fd39-66b8-437b-9194-c1b553f54a8f.png)

**Reply by PZiaugra** [reply]
We still don't have native support for this functionality?
I tried to do according to @Chuxel comment above but I didn't get success.
My scenario is very similar to @wmhilton.

**Comment by mattvb91** [comment]
Would love to have this too. I am working on a multi tenant stack and it requires subdomains. I can forward ports locally etc.. but it would be nice to just have this ability directly inside of codespace

**Comment by waynetheisinger** [comment]
I would also like this - we would love to use codespaces for easy developer onboarding and teaching them how to port forward a bunch of subdomains is another onboarding task... we'd love to just have them be able to intuit how to get up and running with very few instructions

**Comment by neoramos** [comment]
This is something I would like as well.

**Comment by serge-salamanka** [comment]
+1 for this feature. we have a stack of applications that rely on subdomain

**Comment by wjurkowlaniec** [comment]
would be great and allow testing oauth2 apps

**Comment by Drowze** [comment]
We have a multitenant application reliant on subdomains. This is definitely one of the biggest drawbacks we have with Codespaces.

**Comment by matthiasPOE** [comment]
Has anyone managed to make progress on this?

**Comment by rayjlinden** [comment]
I need subdomains as well.  Our dev environment launches docker-compose with things like local stack and elasticsearch - along with our various "micro-services".  Each open on port 443.  We use let's encrypt to issue a wildcard domain cert on the host.  This way each micro-service can talk to each other via ssl.

Not sure how we are going to do this in a Codespace...

**Comment by ghost** [comment]
Open issue, 30+ thumbs up, 0 fix. Classic

**Comment by charlesaraujo** [comment]
+1 for this feature.

**Comment by VissionNZ** [comment]
+1 again! Our Laravel application's tenancy architecture relies on subdomains. Codespaces is great but this is a big stumbling block for us to widely adopt it.

**Comment by Gustavo-Ziaugra-Arvo** [comment]
Its being so long that I dont even belive they will get this
